### PR TITLE
Add distcc support to RunWLLVM

### PR DESCRIPTION
**Pull request by danjujan**
This pull request allows RunWLLVM projects to opt for distributed compilation with distcc.

### CODE REVIEW

File: varats-core/varats/experiment/wllvm.py [hunk 1]
```diff
@@ -61,8 +62,39 @@ class RunWLLVM(base.Extension):  # type: ignore
...
+
+    The distcc_hosts argument will instruct WLLVM to call distcc instead of
+    clang directly. It is the user's responsibility to ensure that all
+    distcc hosts are configured to use exactly the same compiler as localhost!
     """
 
+    def __init__(
+        self,
+        distcc_hosts: tp.Optional[str] = None,
+        *extensions: 'Extension',
```
Comment (vulder) [inline varats-core/varats/experiment/wllvm.py]:
```suggestion
        *extensions: base.Extension,
```

Comment (vulder) [inline varats-core/varats/experiment/wllvm.py]: Defaulted parameters are normally placed at the end, is this not possible for this?

Also, what does this mean for the API surface? Could you please add 2 examples to the doc string (one without and one with) that show how one would use this class.

File: varats-core/varats/experiment/wllvm.py [hunk 2]
```diff
@@ -61,8 +62,39 @@ class RunWLLVM(base.Extension):  # type: ignore
...
+        config: tp.Optional[tp.Dict[str, str]] = None,
+        **kwargs: tp.Any
+    ):
+        super().__init__(*extensions, config=config, **kwargs)
+        self._distcc_hosts = distcc_hosts
+
+    def __create_distcc_scripts(self) -> tp.Tuple[Path, Path]:
+        projdir = Path(sys.argv[0]).parent
```
Comment (vulder) [inline varats-core/varats/experiment/wllvm.py]:
```suggestion
        project_dir = Path(sys.argv[0]).parent
```
This is the more common spelling in the code base

**Review by vulder — CHANGES_REQUESTED** [review]
Overall, this look like a nice addition for people that want to use distcc to speed up the project compilation.

thx for contributing this

File: varats-core/varats/experiment/wllvm.py (hunk 1 above)
Comment (danjujan) [reply varats-core/varats/experiment/wllvm.py]: Fixed in 3e5d671

Comment (danjujan) [reply varats-core/varats/experiment/wllvm.py]: It is possible. Placed it at the end and added examples in 3e5d671.

File: varats-core/varats/experiment/wllvm.py (hunk 2 above)
Comment (danjujan) [reply varats-core/varats/experiment/wllvm.py]: Fixed in 3e5d671

**Review by vulder — APPROVED** [review]

**Pull request merged**